inbox/headless/api-reference.mdx

```diff
@@ -0,0 +1,331 @@
+---
+title: 'Headless Inbox API Reference'
+sidebarTitle: 'API Reference'
+---
+## Novu
+
+`Novu` is is the top level class that is used to interact with the Novu API client.
+
+### Prarams
+
+<ParamField path="applicationIdentifier" type="string" required>
+`applicationIdentifier` is a unique identifier for the application. This is is public credential that is used to authenticate the application with the Novu API. `applicationIdentifier` can be copied from [API Keys](https://dashboard.novu.co/api-keys) page.
+</ParamField>
+
+<ParamField path="subscriberId" type="string" required>
+  `subscriberId` is a unique identifier for the subscriber. Read more about [subscribers](/concepts/subscribers).
+</ParamField>
+
+<ParamField path="subscriberHash" type="string">
+  `subscriberHash` is a unique identifier for the subscriber. It is used for HMAC encryption. Read more about [HMAC Encryption](/react/advanced-configuration#hmac-encryption).
+</ParamField>
+
+<ParamField path="backendUrl" type="string" default="https://api.novu.co">
+Use `https://eu.api.novu.co` value for EU region.
+</ParamField>
+
+<ParamField path="socketUrl" type="string" default="https://ws.novu.co">
+  Use `https://eu.ws.novu.co` value for EU region.
+</ParamField>
+
+### Usage
+<Tabs>
+<Tab title="US" >
+```typescript
+import { Novu } from "@novu/js";
+
+const novu = new Novu({
+  subscriberId: "SUBSCRIBER_ID",
+  applicationIdentifier: "APPLICATION_IDENTIFIER",
+});
+```
+</Tab>
+<Tab title="EU">
+```typescript
+import { Novu } from "@novu/js";
+
+const novu = new Novu({
+  subscriberId: "SUBSCRIBER_ID",
+  applicationIdentifier: "APPLICATION_IDENTIFIER",
+  backendUrl: "https://eu.api.novu.co",
+  socketUrl: "https://eu.ws.novu.co",
+});
+```
+</Tab>
+<Tab title="HMAC Encryption">
+```typescript
+import { Novu } from "@novu/js";
+
+const novu = new Novu({
+  subscriberId: "SUBSCRIBER_ID",
+  applicationIdentifier: "APPLICATION_IDENTIFIER",
+  subscriberHash: "SUBSCRIBER_HASH_HMAC_ENCRYPTIon",
+});
+```
+</Tab>
+</Tabs>
+
+### Methods
+
+- [Notifications](#notifications)
+- [Preferences](#preferences)
+
+## Notifications
+
+### Methods
+- [list](#list)
+- [count](#count)
+- [read](#read)
+- [unread](#unread)
+- [archive](#archive)
+- [unarchive](#unarchive)
+- [readAll](#readAll)
+- [archiveAll](#archiveAll)
+- [archiveAllRead](#archiveAllRead)
+- [completePrimary](#completePrimary)
+- [completeSecondary](#completeSecondary)
+- [revertPrimary](#revertPrimary)
+- [revertSecondary](#revertSecondary)
+
+### list
+`list` method fetch the list of notifications for the subscriber based on the provided filters.
```

Comment:
We should probably talk about the caching here and how it works in case of fetching more.

Reply:
Can you please help me with caching details? how it works?